### ISSUE TITLE
Fresh install - docs link returning a 404

### ISSUE THREAD

**Issue by aronasorman** [learningequality/ka-lite#5069]
## Summary

I get a 404 page when visiting this link: `/static/html/usermanual/userman_main.html` (that's the doc page).

Possible cause: we ignore `html` files when we run collectstatic: https://github.com/learningequality/ka-lite/blob/master/kalite/distributed/management/commands/setup.py#L480

Workaround: Run collectstatic again.

**Comment by aronasorman** [comment]
Not sure what the best solution is here. Easiest is to just disable that docs link.

**Comment by radinamatic** [comment]
Is that with the build 210 @aronasorman? I had some red error code appear in the last Command Prompt window while installing fresh on Windows 7, and I can't start the server form task tray.

Going to test again, see if it repeats...

**Comment by aronasorman** [comment]
This is with an sdist with the latest 0.16.x @radinamatic. Can you try it with an installer? 211 is what you want.

**Comment by radinamatic** [comment]
Gee, Bamboo & you are popping those builds like mushrooms after rain... :grinning: 
I just downloaded 210, and there's already 211!

**Comment by radinamatic** [comment]
Btw, build 210 repeatedly failed to install correctly at the last step with `pip.exe`, no `kalite` nor `kalite.bat`files where they should be, so server couldn't start:

![ie11 - win7 -build-210](https://cloud.githubusercontent.com/assets/1457929/14228911/e411e776-f925-11e5-8400-b80840b2161f.png)

Will test with 211 now!

**Comment by aronasorman** [comment]
Did you see some kind of traceback @radinamatic?

**Comment by radinamatic** [comment]
Command Prompt closes too fast to see anything... :(

**Comment by radinamatic** [comment]
@aronasorman Had the same fail with build 211 on both Windows XP & 8.1, as previously with 210 on Win 7. 

The last step, when Command Prompt opens after message "...installer will now copy kalite source into Python site-packages..." or similar, 5-6 lines of red output too fast to read, Command Prompt closes and installer GUI reports installation has finished, even though no `kalite` nor `kalite.bat` files where they should be, so server can't start.

**Comment by aronasorman** [comment]
Gotcha. Let's wait for @MCGallaspy to provide input on the windows failure and the proper fix for the docs.

**Comment by radinamatic** [comment]
Installer script for Windows did not change since build 208, so something else must have been moved around to impede those last steps to complete correctly, right? @aronasorman

**Comment by aronasorman** [comment]
If the last message is indeed right, then there error should be around here: https://github.com/learningequality/installers/blob/develop/windows/installer-source/KaliteSetupScript.iss#L377-L382

**Comment by radinamatic** [comment]
True, and it would be more helpful if the user actually had time to see this message... :stuck_out_tongue:

**Comment by radinamatic** [comment]
I just tested the latest OSX installer from Bamboo (build 111), and the link in docs tab works fine and opens the `http://127.0.0.1:8008/static/html/usermanual/userman_main.html` without problems. 

It is not possible to test this on the latest Windows installer (build 211) because of the error explained above (there's neither `kalite.bat` nor `.kalite/` folder), but I just tried a VM where I had build 208 installed 4 days ago, which otherwise worked fine, and it was not possible to open docs either. All folders inside `.kalite\httpsrv\static\html\` are empty: 

![ie9 - win7 0 16 build 208 running - oracle vm virtualbox_420](https://cloud.githubusercontent.com/assets/1457929/14235332/ba751d58-f9fb-11e5-8b52-100690734f2c.png)

**Comment by radinamatic** [comment]
As expected, installer `ka-lite-bundle_0.16.0-0ubuntu1_all.deb` cannot open the docs, no `html` folder at all:

![selection_421](https://cloud.githubusercontent.com/assets/1457929/14235537/9140f190-fa01-11e5-81f3-678dae5b837c.png)

**Comment by djallado** [comment]
I tested also this also using OS X installer from dungeon (build 111) on OS X El Capitan and the `DOCS` link works fine. 
Here's my screenshot:
![doclink](https://cloud.githubusercontent.com/assets/8664071/14242150/568a3de6-fa82-11e5-8c29-9ebe9dc16afc.png)

**Comment by djallado** [comment]
I tested Windows installer (build 211) from dungeon. I can't start the server.

**Comment by MCGallaspy** [comment]
Windows installer should work again, I think. Needed to update an env variable (`KALITE_BUILD_VERSION=0.16.1`) on the build server. @aronasorman have you tried running collectstatic again to resolve this issue?

**Comment by aronasorman** [comment]
@MCGallaspy yup, running collect static again resolves this issue. Should
we have the installers rerun said command to make it work?

**Comment by radinamatic** [comment]
Windows installer works OK with build 212! :+1: @MCGallaspy @aronasorman

**Comment by radinamatic** [comment]
I mean, docs are not opening, but it installs a-OK! :wink:

**Comment by MCGallaspy** [comment]
> Should we have the installers rerun said command to make it work?

No, that would violate our rule of minimizing the responsibility of the installers -- it's a bug in the application. And I think that running `collectstatic` twice would simply re-introduce the bug it was intended to fix, unless `collectstatic` updates the timestamp of the files it touches. Check out the commit message I wrote for that fix:

```
Force clear out collectstatic on setup
In some upgrade scenarios (like on the RPi) the system's
built js files might have a timestamp *earlier* than the
static files appearing in the user's static files location.
The effect is that `collectstatic` called during an upgrade
will not replace the old static files with the new ones, since
the timestamp is reverse.

So force them to be cleared. But preserve the "extra" files we
add in, e.g. for subtitles (vtt, srt) and exercise templates (html).
Those must not be removed, since they're downloadable content!
```

We're ignoring the docs html files (they're static files). If the timestamp of the js files are indeed updated on copying, then we can run collecstatic again without the clear and ignore options. However if the timestamps are not changed then it will simply reintroduce the bug. In that case we'll have to figure out another solution to the above issues -- perhaps by touching the system's static files on upgrade. Since I'm travelling at the moment, if this is urgently needed for release, can someone else take it over?

**Comment by aronasorman** [comment]
@MCGallaspy I'm making the fix now. I'll test on the RPi to see if the timestamps are updated.

**Comment by aronasorman** [comment]
Fixed in #5074.
